### PR TITLE
TIP-706: Add product status PQB sorter

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

This PR adds a sorter on product status (enabled).

## ES Sorter Checklist

- [x] Add sorter integration tests for the field in the PQB
- [ ] ~Add missing integration to the PimCatalogXIntegration tests for sort~ Not needed
- [x] Add the sorter + specs
- [x] Update the reference documentation

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst [hunk 1]
```diff
@@ -345,6 +345,18 @@ Not Equal (!=)
...
 
+Sorting
+~~~~~~~
+
+.. code-block:: php
+
+    'sort' => [
+        'enabled' => [
+            'order'   => 'asc',
+            'missing' => '_first'
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst]: _last

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/StatusSorterIntegration.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        if (1 === self::$count || $this->getConfiguration()->isDatabasePurgedForEachTest()) {
+            $this->createProduct('foo', ['enabled' => false]);
+            $this->createProduct('bar', ['enabled' => true]);
+            $this->createProduct('baz', ['enabled' => false]);
+            $this->createProduct('foobar', ['enabled' => true]);
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/StatusSorterIntegration.php:26]: Add one without an enabled value

**Review by SamirBoulil — APPROVED** [review]
MOB GTM review

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/StatusSorterIntegration.php (hunk 2 above)
Comment (damien-carcel) [reply src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/StatusSorterIntegration.php:26]: Done with `foobaz`.

**Pull request merged**